shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -341,6 +341,12 @@ public FlutterRenderer getRenderer() {
     return renderer;
   }
 
+  /** The FlutterJNI associated with this engine. */
+  @NonNull
+  public FlutterJNI getFlutterJNI() {
```

Comment:
The `FlutterJNI` should be private to `FlutterEngine`.

For other built-in plugins/channels, the `FlutterEngine` constructs the plugin and passes it the `FlutterJNI` if needed.  `FlutterEngine` can then provide an accessor to make the plugin available to `FlutterView`.

Reply:
Refactored to have the `FlutterEngine` construct plugin as described.